File: src/main/java/cn/dmego/leetcode/solution/lc1_lc50/Solution_1.java
package cn.dmego.leetcode.solution.lc1_lc50;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

/**
 * 两数之和
 * @author dmego
 * @date 2021/12/30 09:36
 */
public class Solution_1 {

    public int[] twoSum(int[] nums, int target) {
        if (nums.length < 2) return new int[]{};
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                return new int[]{i, map.get(target - nums[i])};
            } else {
                map.put(nums[i], i);
            }
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        int[] nums = new int[]{3,2,4};
        int target = 6;
        Solution_1 lc = new Solution_1();
        int[] result = lc.twoSum(nums, target);
        System.out.println(Arrays.toString(result));
    }

}

File: python/leetcode/solution/Solution_1.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
"""
@Date    ：2024/12/31 15:11
@File    ：Solution_1.py
@Author  ：Dmego
"""

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) < 2: return []
        nums_dict = {}
        for i, num in enumerate(nums):
            other = target - num
            if other in nums_dict:
                return [nums_dict[other], i]
            else:
                nums_dict[num] = i
        return []



def main():
    so = Solution()
    nums = [2,7,11,15]
    target = 9
    result = so.twoSum(nums, target)
    print(result)

if __name__ == '__main__':
    main()